src/SentTimerTeensyQuad.h

```diff
@@ -0,0 +1,163 @@
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_B0_00,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_B0_01,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_B0_02,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_B0_03,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_B0_04,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_B0_05,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_AD_B1_00,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_B0_06,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_AD_B1_01,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_B0_07,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_AD_B1_02,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_B0_08,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_AD_B1_03,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_B0_09,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_B0_10,
+    &IOMUXC_SW_MUX_CTL_PAD_GPIO_B0_11,
+};
+
+unsigned char ioMuxValue[numPins] {
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+};
+
+volatile uint32_t *ioSelCtrl[numPins] = {
+    nullptr,
+    nullptr,
+    nullptr,
+    &IOMUXC_QTIMER2_TIMER0_SELECT_INPUT,
+    &IOMUXC_QTIMER2_TIMER1_SELECT_INPUT,
+    &IOMUXC_QTIMER2_TIMER2_SELECT_INPUT,
+    &IOMUXC_QTIMER3_TIMER0_SELECT_INPUT,
+    &IOMUXC_QTIMER3_TIMER0_SELECT_INPUT,
+    &IOMUXC_QTIMER3_TIMER1_SELECT_INPUT,
+    &IOMUXC_QTIMER3_TIMER1_SELECT_INPUT,
+    &IOMUXC_QTIMER3_TIMER2_SELECT_INPUT,
+    &IOMUXC_QTIMER3_TIMER2_SELECT_INPUT,
+    &IOMUXC_QTIMER3_TIMER3_SELECT_INPUT,
+    nullptr,
+    nullptr,
+    nullptr,
+};
+
+unsigned char ioSelValue[numPins] {
+    0, 0, 0, 1, 1, 1, 1, 2, 0, 2, 1, 2, 1, 0, 0, 0,
+};
+
+volatile IMXRT_TMR_t *quadTimers[4] = {
+    (IMXRT_TMR_t *)IMXRT_TMR1_ADDRESS, 
+    (IMXRT_TMR_t *)IMXRT_TMR2_ADDRESS, 
+    (IMXRT_TMR_t *)IMXRT_TMR3_ADDRESS, 
+    (IMXRT_TMR_t *)IMXRT_TMR4_ADDRESS
+};
+
+void isr_tmr1();
+void isr_tmr2();
+void isr_tmr3();
+void isr_tmr4();
+
+void (*isrFuncs[4])() = {isr_tmr1, isr_tmr2, isr_tmr3, isr_tmr4};
+IRQ_NUMBER_t irqNumbers[4] = {IRQ_QTIMER1, IRQ_QTIMER2, IRQ_QTIMER3, IRQ_QTIMER4};
+void *listeners[4][4] = {nullptr};
+
+class SentTimerTeensyQuad:public BaseSent{
+    public:
+        SentTimerTeensyQuad(uint8_t tick_time = 3, bool padding = false, unsigned int pin = 4)
+            : BaseSent(tick_time, padding, sentBuffer)
+            , _pin{pin}
+        {}
+
+        void begin(SentCallback callback = nullptr) override {
+            bool isValid;
+            unsigned char pinIndex;
+            for(pinIndex=0; pinIndex<sizeof(validPins); pinIndex++) {
+                if(_pin == validPins[pinIndex]) {
+                    quadTimer = quadTimers[tmrIndex[pinIndex]-1];
+                    quadTimerCh = &(quadTimer->CH[chnIndex[pinIndex]]);
+                    isValid = true;
+                    break;
+                }
+            }
+
+            if(!isValid) {
+                return;
+            }
+
+            pinMode(_pin, INPUT);
+            *ioMuxCtrl[pinIndex] = ioMuxValue[pinIndex];
+            if(ioSelCtrl[pinIndex] != nullptr) *ioSelCtrl[pinIndex] = ioSelValue[pinIndex];
+
+            attachInterruptVector(irqNumbers[tmrIndex[pinIndex]-1], isrFuncs[tmrIndex[pinIndex]-1]);
```

Comment:
Not sure about the `tmrIndex[...]-1`. 

You probably go for the TimerX where the X is in the array. But now you have the -1 everywhere. I guess this needs a code comment explaining the situation.

Reply:
Sadly NXP uses 1-4 for timers and 0-3 for timer channels... The numbering matches the datasheet, which I think makes the most sense. Will add a note.